### PR TITLE
feat(parquet): decimal logic type

### PR DESCRIPTION
**Pull request by belom88**
DECIMAL type can annotate different physical types.
Implemented here:
- [x] int32
- [x] int64
- [x] fixed_len_byte_array (only `fromPrimitive` function)
- [x] binary: precision (only `fromPrimitive` function)

`toPrimitive` functions have't been tested.

### CODE REVIEW

File: modules/parquet/src/parquetjs/schema/types.ts [hunk 1]
```diff
@@ -239,6 +257,16 @@ function toPrimitive_INT32(value: any) {
   return v;
 }
 
+function decimalToPrimitive_INT32(value: number, field: ParquetField) {
```
Comment (dryabinin-actionengine) [inline modules/parquet/src/parquetjs/schema/types.ts:260]: Please add a return type to function.

**Comment by dryabinin-actionengine** [comment]
Please add tests for unblocked files.

**Review by dryabinin-actionengine — APPROVED** [review]

File: modules/parquet/src/parquetjs/schema/declare.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export type ParquetCompression =
   | 'LZ4_RAW'
   | 'ZSTD';
 export type RepetitionType = 'REQUIRED' | 'OPTIONAL' | 'REPEATED';
-export type ParquetType = PrimitiveType | OriginalType | DecimalTypes;
+export type ParquetType = PrimitiveType | OriginalType;
```
Comment (ibgreen) [inline modules/parquet/src/parquetjs/schema/declare.ts]: Why are we removing it here... 

'I.e. what is the difference between ParquetType and PrimitiveType, why do we need both?

Should we add JSDoc to these types explaining the difference?

**Review by ibgreen — APPROVED** [review]

File: modules/parquet/src/parquetjs/schema/declare.ts (hunk 2 above)
Comment (belom88) [reply modules/parquet/src/parquetjs/schema/declare.ts]: `ParquetType` is union of `Primitive` (Physical) and `Original` (Logical) types. The naming and related design came from forked library.

Comment (belom88) [reply modules/parquet/src/parquetjs/schema/declare.ts]: It is change of last commit. I revert the change of first commit because it is unnecessary.

**Pull request merged**